media-video/motion/motion-4.3.1.ebuild

```diff
@@ -0,0 +1,104 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+inherit autotools readme.gentoo-r1 systemd
+
+DESCRIPTION="A software motion detector"
+HOMEPAGE="https://motion-project.github.io"
+SRC_URI="https://github.com/Motion-Project/${PN}/archive/release-${PV}.tar.gz -> ${PN}-release-${PV}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~hppa ~ia64 ~ppc ~ppc64 ~sparc ~x86"
+IUSE="ffmpeg libav mariadb mmal mysql postgres sqlite supervise-daemon v4l webp"
+
+RDEPEND="
+	acct-group/motion
+	acct-user/motion
+	virtual/jpeg:=
+	ffmpeg? (
+		libav? ( media-video/libav:= )
+		!libav? ( media-video/ffmpeg:0= )
+	)
+	mariadb? ( dev-db/mariadb-connector-c )
+	mmal? ( media-libs/raspberrypi-userland )
+	mysql? ( dev-db/mysql-connector-c )
+	postgres? ( dev-db/postgresql:= )
+	sqlite? ( dev-db/sqlite:3 )
+	webp? ( media-libs/libwebp:= )
+"
+DEPEND="${RDEPEND}
+	net-libs/libmicrohttpd
+	v4l? (
+		media-libs/libv4l
+		virtual/os-headers
+	)
+"
+
+DISABLE_AUTOFORMATTING="yes"
+DOC_CONTENTS="
+You need to setup a configuraton file (/etc/motion/motion.conf) before
+running motion for the first time.
+
+If motion is built with the mysql or mariadb use flags then please make
+sure to configure a matching database_type in the config file.
+
+Motion's default current working directory is /var/lib/motion.
+Output will be written here, unless overruled in the config file.
+
+Motion runs by default under user motion and group motion:
+- change this if needed in /etc/conf.d/motion
+- or add users who need access to the output files to the motion group
+
+To install motion as a service, use:
+- rc-update add motion default # with OpenRC
+- systemctl enable motion.service # with systemd
+"
+
+S="${WORKDIR}"/${PN}-release-${PV}
+
+src_prepare() {
+	default
+	eautoreconf
+}
+
+src_configure() {
+	econf \
+		$(use_with ffmpeg) \
+		$(use_with mariadb) \
+		$(use_with mmal) \
+		$(use_with mysql) \
+		$(use_with postgres pgsql) \
+		$(use_with sqlite sqlite3) \
+		$(use_with v4l v4l2) \
+		$(use_with webp) \
+		--without-optimizecpu
+}
+
+src_install() {
+	emake \
+		DESTDIR="${D}" \
+		docdir=/usr/share/doc/${PF} \
+		examplesdir=/usr/share/doc/${PF}/examples \
+		install
+
+	newconfd "${FILESDIR}"/${PN}.confd-r4 ${PN}
+	newinitd "${FILESDIR}"/${PN}.initd-r4 ${PN}
+
+	# do not start mysql/postgres unnecessarily
+	( use mysql || use mariadb ) || sed -i -e "/use mysql/d"  "${D}/etc/init.d/motion"
+	use postgres || sed -i -e "/use postgresql/d"  "${D}/etc/init.d/motion"
+	( use mysql || use mariadb || use postgres ) || sed -i -e "/depend/{N;N;d;}" "${D}/etc/init.d/motion"
+
+	# configure for supervise-daemonn
+	use supervise-daemon && echo "supervisor=\"supervise-daemon\"" >> "${D}/etc/init.d/motion"
+	use supervise-daemon || echo "pidfile=\"/run/\${RC_SVCNAME}.pid\"" >> "${D}/etc/init.d/motion"
+	use supervise-daemon || echo "command_background=true" >> "${D}/etc/init.d/motion"
```

Comment:
This is doable with a single echo. Although I'd say to construct this file with `newinitd`.

Reply:
I tried doing with a single echo, using `echo -e`, and adding a `\n` for the newline but that added unwanted spaces. An alternative would be using a heredoc, but that won't be shorter nor easier to read, I think.

Not sure how to construct a file using newinitd, other then have a separate initd file for this, which would make us maintain more files.